### PR TITLE
enhance: Make dynamic load/release partition follow targets

### PR DESCRIPTION
**Pull request by congqixia**
Related to #37849

### CODE REVIEW

File: internal/querycoordv2/job/job_sync.go [hunk 1]
```diff
@@ -89,12 +89,27 @@ func (job *SyncNewCreatedPartitionJob) Execute() error {
...
+	err := job.meta.CollectionManager.PutPartition(job.ctx, partition)
 	if err != nil {
 		msg := "failed to store partitions"
 		log.Warn(msg, zap.Error(err))
 		return errors.Wrap(err, msg)
 	}
 
-	return nil
+	// manual trigger update next target
+	ready, err := job.targetObserver.UpdateNextTarget(req.GetCollectionID())
```
Comment (czs007) [inline internal/querycoordv2/job/job_sync.go:100]: Compared to before, awaiting the next target here, if the next target includes an accompanying segment change, the execution time here will be prolonged, potentially affecting the time taken by RootCoord to execute create_partition

**Comment by xiaofan-luan** [comment]
good job, this looks to be much cleaner to me.

The major draw back czs point out is ok to me

**Comment by congqixia** [comment]
rerun go-sdk

**Pull request merged**